src/views/product-root-docs-path-landing/server.ts

```diff
@@ -134,19 +132,31 @@ const getStaticProps = async (context: GetStaticPropsContext) => {
 			basePath,
 			baseName,
 		})
-
-	// TODO: replace any with accurate type
-	const generatedProps = (await generatedGetStaticProps({
+	const getStaticPropsResult = (await generatedGetStaticProps({
 		...context,
 		params: { page: [] },
-	})) as $TSFixMe
+		// TODO: surely there isn't a need to cast here... but there seems to be?
```

Comment:
💭 I would love to clean this type up, but got kind of stuck with it... any and all input welcome!

I'm going to try to dig a bit more into NextJS + TypeScript, it seems odd to me that this requires casting since I've set what I _thought_ was the correct return type on the function that returns `generatedGetStaticProps` 🤔 ... but which must not be correct since this isn't working 😅

Reply:
I think I figured this out - I needed to account for `redirect` and `notFound` cases. Have added a little conditional to handle those cases, which conveniently also serves as a type guard.